app/src/main/java/de/westnordost/streetcomplete/quests/memorial_type/AddMemorialType.kt

```diff
@@ -0,0 +1,45 @@
+package de.westnordost.streetcomplete.quests.memorial_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement
+
+class AddMemorialType : OsmFilterQuestType<String>() {
+
+    override val elementFilter = """
+        nodes, ways, relations with
+          historic=memorial
+          and (!memorial or memorial=yes)
+          and !memorial:type
+    """
+    override val changesetComment = "Add memorial type"
+    override val wikiLink = "Key:memorial"
+    override val icon = R.drawable.ic_quest_memorial
+    override val questTypeAchievements = listOf(QuestTypeAchievement.CITIZEN)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_memorialType_title
+
+    override fun createForm() = AddMemorialTypeForm()
+
+    override fun applyAnswerTo(answer: String, tags: Tags, timestampEdited: Long) {
+        when(answer) {
```

Comment:
Off-topic and nitpick, but I noticed you often forget to add a space after `if` and `when`. Have you tried the Ktlint plugin for Android Studio (linked in the [contributing guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md?plain=1#L107))? That may help familiarizing with the style.

Reply:
> Have you tried the Ktlint plugin for Android Studio (linked in the [contributing guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md?plain=1#L107))?

no, I installed it now